vertical-pod-autoscaler/pkg/recommender/model/cluster_test.go

```diff
@@ -82,6 +82,59 @@ func TestClusterGCAggregateContainerStateDeletesOld(t *testing.T) {
 	assert.Empty(t, vpa.aggregateContainerStates)
 }
 
+func TestClusterGCAggregateContainerStateDeletesOldEmpty(t *testing.T) {
+	// Create a pod with a single container.
+	cluster := NewClusterState()
+	vpa := addTestVpa(cluster)
+	addTestPod(cluster)
+
+	assert.NoError(t, cluster.AddOrUpdateContainer(testContainerID, testRequest))
+	// No usage samples added.
+
+	assert.NotEmpty(t, cluster.aggregateStateMap)
+	assert.NotEmpty(t, vpa.aggregateContainerStates)
+
+	assert.Len(t, cluster.aggregateStateMap, 1)
+	var creationTime time.Time
+	for _, aggregateState := range cluster.aggregateStateMap {
+		creationTime = aggregateState.CreationTime
+	}
+
+	// AggegateContainerState are valid for 8 days since creation
+	cluster.GarbageCollectAggregateCollectionStates(creationTime.Add(9 * 24 * time.Hour))
```

Comment:
Please do a clean up with smaller timestamp, assert the aggregation isn't cleaned up just because it lacks active pods, then do clean up with creationTime + 9 days to check that it's cleaned up because it's old.

Reply:
Done